src/rofl/datapath/pipeline/openflow/openflow1x/pipeline/of1x_match.c

```diff
@@ -1557,47 +1557,71 @@ void __of1x_destroy_match_group(of1x_match_group_t* group){
 
 
 
-void __of1x_match_group_push_back(of1x_match_group_t* group, of1x_match_t* match){
+rofl_result_t __of1x_match_group_insert(of1x_match_group_t* group, of1x_match_t* match){
+	of1x_match_t* iter = NULL;
 
-	if ( unlikely(group==NULL) || unlikely(match==NULL) )
-		return;
+ 	if ( unlikely(group==NULL) || unlikely(match==NULL) )
+		return ROFL_FAILURE;
 
 	match->next = match->prev = NULL; 
 
+	//Insert in array
+	if ( group->m_array[match->type] != NULL ){
```

Comment:
Before accessing the array `match->type` should be checked to be strictly smaller than `OF1X_MATCH_MAX` to prevent overflows.